packages/vuetify/src/components/VBadge/VBadge.ts

```diff
@@ -1,72 +1,202 @@
+      }
+
+      return this.left ? 'auto' : this.computedXOffset
+    },
+    computedRight (): string {
+      if (this.isRtl) {
+        return this.left ? 'auto' : this.computedXOffset
+      }
+
+      return !this.left ? 'auto' : this.computedXOffset
+    },
+    computedTop (): string {
+      return !this.bottom ? 'auto' : this.computedYOffset
+    },
+    computedXOffset (): string {
+      return this.calcPosition(this.offsetX)
+    },
+    computedYOffset (): string {
+      return this.calcPosition(this.offsetY)
+    },
+    isRtl (): boolean {
+      return getObjectValueByPath(this.$vuetify, 'rtl', false)
+    },
+    // Default fallback if offsetX
+    // or offsetY are undefined.
+    offset (): number {
+      if (this.overlap) return this.dot ? 8 : 12
+      return this.dot ? 2 : 4
+    },
+    styles (): object {
+      if (this.inline) return {}
+
+      return {
+        bottom: this.computedBottom,
+        left: this.computedLeft,
+        right: this.computedRight,
+        top: this.computedTop,
       }
     },
   },
 
-  render (h): VNode {
-    const badge = this.$slots.badge && [h('span', this.setBackgroundColor(this.color, {
-      staticClass: 'v-badge__badge',
-      attrs: this.$attrs,
-      directives: [{
-        name: 'show',
-        value: this.isActive,
-      }],
-    }), this.$slots.badge)]
+  methods: {
+    calcPosition (offset: string | number): string {
+      return `calc(100% - ${convertToUnit(offset || this.offset)})`
+    },
+    genBadge () {
+      const fallback = { t: (text = '') => text }
+      const lang = getObjectValueByPath(this.$vuetify, 'lang', fallback)
+      const label = this.$attrs['aria-label'] || lang.t(this.label)
 
-    return h('span', {
-      staticClass: 'v-badge',
-      class: this.classes,
-    }, [
-      this.$slots.default,
-      this.transition ? h('transition', {
+      const data = this.setBackgroundColor(this.color, {
+        staticClass: 'v-badge__badge',
+        style: this.styles,
+        attrs: {
+          'aria-atomic': this.$attrs['aria-atomic'] || 'true',
+          'aria-label': label,
+          'aria-live': this.$attrs['aria-live'] || 'polite',
+          title: this.$attrs.title,
+          role: this.$attrs.role || 'status',
+        },
+        directives: [{
+          name: 'show',
+          value: this.isActive,
+        }],
+      })
+
+      const badge = this.$createElement('span', data, [this.genBadgeContent()])
+
+      if (!this.transition) return badge
+
+      return this.$createElement('transition', {
         props: {
           name: this.transition,
           origin: this.origin,
           mode: this.mode,
         },
-      }, badge) : badge,
-    ])
+      }, [badge])
+    },
+    genBadgeContent () {
+      const slot = getSlot(this, 'badge')
+
+      if (slot) return slot
+      // Dot prop shows no content
+      if (this.dot) return undefined
+      if (this.content) return String(this.content)
```

Comment:
`When pushing into render children arrays , should either be the vnode or undefined`

I've added "string" in notion

Reply:
Thank you, good catch.